### ISSUE TITLE
how to add the Regularization term in matconvnet loss function?

### ISSUE THREAD

**Issue by eroszip** [vlfeat/matconvnet#887]
as we known, Regularization term is a good way to preventing overfit，in caffe we can easily set the parameter with weight_decay,but how to set it with matconvnet?

**Comment by AruniRC** [comment]
Hi, 
 MatConvNet also has a weight-decay term.
 
 Each layer has a weight decay field (or net.param in case of DAG has a field called net.param(i).weightDecay).
 This is usually set to 1 for all layers/parameters.

 Now during cnn_train, a global weight decay term is set (usually 0.0005 by default, hardcoded in the cnn_train code). So for the i-th layer, weight decay becomes net.param(i).weightDecay * 0.0005.

HTH

**Comment by layumi** [comment]
@eroszip 
I think you can easily add "opts.weightDecay = 0.0005;" (Line22 /matlab/cnn_dagnn.m) without changing the weightDecay for every params. 
This opts will multiply the param's weightDecay as the final weightDecay to update the network. 
(Line295 /matlab/cnn_dagnn.m)
